### PR TITLE
Fixes #111 for master and make it Plone 6 only

### LINKED ISSUE #111: querystring-search fails when using SearchableText and special values in search
When using words like and or or in a query to the @querystring-search endpoint that searches in the SearchableText index, the endpoint breaks because of some ZCTextIndex error and returns a HTTP 500.

This is the traceback in Plone:

```
2022-07-07 15:25:55,862 ERROR   [Zope.SiteErrorLog:22][waitress-2] ParseError: http://localhost:3000/en/global-search/@querystring-search
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 371, in publish_module
  Module ZPublisher.WSGIPublisher, line 266, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 63, in call_object
  Module plone.rest.service, line 22, in __call__
  Module plone.restapi.services, line 19, in render
  Module plone.restapi.services.querystringsearch.get, line 56, in reply
  Module plone.app.querystring.querybuilder, line 89, in __call__
  Module plone.app.querystring.querybuilder, line 170, in _makequery
  Module Products.CMFPlone.CatalogTool, line 468, in searchResults
  Module Products.ZCatalog.ZCatalog, line 625, in searchResults
  Module Products.ZCatalog.Catalog, line 1091, in searchResults
  Module Products.ZCatalog.Catalog, line 634, in search
  Module Products.ZCatalog.Catalog, line 564, in _search_index
  Module Products.ZCTextIndex.ZCTextIndex, line 209, in query_index
  Module Products.ZCTextIndex.QueryParser, line 153, in parseQuery
  Module Products.ZCTextIndex.QueryParser, line 193, in _parseOrExpr
  Module Products.ZCTextIndex.QueryParser, line 211, in _parseAndExpr
  Module Products.ZCTextIndex.QueryParser, line 233, in _parseNotExpr
  Module Products.ZCTextIndex.QueryParser, line 241, in _parseTerm
  Module Products.ZCTextIndex.QueryParser, line 260, in _parseAtom
  Module Products.ZCTextIndex.QueryParser, line 188, in _get
  Module Products.ZCTextIndex.QueryParser, line 174, in _require
Products.ZCTextIndex.ParseTree.ParseError: Token <Token:ATOM> required, 'and' found
```

This is in Plone 5.2.

### PR DESCRIPTION
**Pull request by erral**
As per comments in #112

### CODE REVIEW

**Comment by erral** [comment]
@jenkins-plone-org please run jobs

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

File: plone/app/querystring/querybuilder.py [hunk 1]
```diff
@@ -189,33 +206,28 @@ def number_of_results(self, query):
...
         )
 
     def filter_query(self, query):
-        text = query.get('SearchableText', None)
+        text = query.get("SearchableText", None)
         if isinstance(text, dict):
-            text = text.get('query', '')
+            text = text.get("query", "")
         if text:
-            query['SearchableText'] = self.munge_search_term(text)
```
Comment (petschki) [inline plone/app/querystring/querybuilder.py:203]: I'd do the same way as for `1.x` keep the class method for better customization of the searchterm handling.

File: plone/app/querystring/querybuilder.py [hunk 2]
```diff
@@ -189,33 +206,28 @@ def number_of_results(self, query):
...
+            query["SearchableText"] = munge_search_term(text)
         return query
 
-    def munge_search_term(self, q):
```
Comment (petschki) [inline plone/app/querystring/querybuilder.py:206]: keep this and return:
```
    def munge_search_term(self, q):
        return munge_search_term(q)
```

**Review by petschki — CHANGES_REQUESTED** [review]

**Comment by petschki** [comment]
failing robot tests are unrelated

**Pull request merged**